src/main/kotlin/org/sert2521/infiniterecharge2020/colorwheelspin/ColorWheelSpinCommands.kt

```diff
@@ -0,0 +1,93 @@
+package org.sert2521.infiniterecharge2020.colorwheelspin
+
+import edu.wpi.first.wpilibj.util.Color
+import java.lang.System.currentTimeMillis
+import kotlin.math.abs
+import kotlinx.coroutines.cancel
+import org.sert2521.sertain.coroutines.doAll
+import org.sert2521.sertain.events.onTick
+import org.sert2521.sertain.subsystems.doTask
+import org.sert2521.sertain.subsystems.use
+
+suspend fun retract(time: Long) = doTask {
+    val colorWheelSpin = use<ColorWheelSpin>()
+    val startTime = currentTimeMillis()
+    action {
+        onTick {
+            colorWheelSpin.extend(-0.2)
+            if (currentTimeMillis() - startTime < time) {
+                this@action.cancel()
+            }
+        }
+    }
+}
+
+suspend fun spinToColor() = doTask {
```

Comment:
![Capture](https://user-images.githubusercontent.com/23201789/74090671-ebb54f80-4a62-11ea-877c-a68b714d239a.PNG)

Depending on where our onboard color sensor is, the color readings we'll get may not be the correct ones for the field. The sensor on the Control Panel is located towards the edge of the middle